src/lib/genesis_ledger/genesis_ledger.ml

```diff
@@ -62,6 +66,20 @@ module Balances (Balances : Intf.Named_balances_intf) = struct
 end
 
 module Utils = struct
+  (* Populate a [Ledger.Root.t] ledger with the components of a root
+     backing_ledger for a genesis ledger. These components are the underlying
+     root ledger and the stored mask on top of that root that is presented to
+     users of the genesis root. The mask is passed in to this function so we can
+     assert that there are no uncommitted changes to it, allowing us to use the
+     potentially much faster Ledger.Root.transfer_accounts_with function. *)
+  let populate_root_with_backing_root genesis_mask ~src ~dest =
+    assert (
+      Ledger_hash.equal
+        (Ledger.merkle_root genesis_mask)
+        (Ledger.Root.merkle_root src) ) ;
+    Ledger.Root.transfer_accounts_with
+      ~stable:Ledger_transfer_stable.transfer_accounts ~src ~dest
```

Comment:
This *shouldn't* represent a change in behaviour, but this is an extra assertion where there wasn't one before. I'm doing this so the same `val t : Ledger.t Lazy.t` definition can still exist in the `Genesis_ledger`, while still being able to populate a root ledger with `Ledger.Root.transfer_accounts_with`. Ideally we'd have some notion of read-only ledger and have the genesis ledger be one of those, but that doesn't yet exist.